src/config/logger.ts

```diff
@@ -77,44 +76,53 @@ export const overrideProbotLoggingMethods = (probotLogger: Logger) => {
 		type: "stream",
 		stream: LOG_STREAM,
 		closeOnExit: false,
-		level: globalLoggingLevel
+		level: defaultLogLevel
 	});
 };
 
-const createNewLogger = (name: string, options: Partial<LoggerOptions> = {}): Logger => {
-	return createLogger({
+const createNewLogger = (name: string, options: LoggerOptions = {}): Logger => {
+	return createLogger(merge<Logger.LoggerOptions, LoggerOptions>({
 		name,
 		stream: LOG_STREAM,
-		level: globalLoggingLevel,
+		level: defaultLogLevel,
 		serializers: {
 			err: errorSerializer,
 			error: errorSerializer,
 			res: responseSerializer,
 			response: responseSerializer,
 			req: requestSerializer,
 			request: requestSerializer
-		},
-		...options
-	});
+		}
+	}, options));
 };
 
-export const getLogger = (name: string, fields?: Record<string, unknown>): Logger => {
-	const logger = createNewLogger(name);
-	logger.addSerializers({
-		jiraHost: hashSerializer,
-		orgName: hashSerializer,
-		repoName: hashSerializer,
-		userGroup: hashSerializer,
-		aaid: hashSerializer,
-		username: hashSerializer
-	});
-	return logger.child({ ...fields });
+interface LoggerOptions {
+	fields?: Record<string, unknown>;
+	streams?: Stream[];
+	level?: LogLevel;
+	stream?: NodeJS.WritableStream;
+	serializers?: Serializers;
+	src?: boolean;
+}
+
+export const getLogger = (name: string, options: LoggerOptions = {}): Logger => {
+	const logger = createNewLogger(name, merge({
+		serializers: {
+			jiraHost: hashSerializer,
+			orgName: hashSerializer,
+			repoName: hashSerializer,
+			userGroup: hashSerializer,
+			aaid: hashSerializer,
+			username: hashSerializer
+		}
+	}, options));
+	return options.fields ? logger.child({ ...options.fields }) : logger;
 };
 
 // This will log data to a restricted environment [env]-unsafe and not serialize sensitive data
-export const getUnsafeLogger = (name: string, fields?: Record<string, unknown>): Logger => {
-	const logger = createNewLogger(name, { env_suffix: "unsafe" });
-	return logger.child({ ...fields });
+export const getUnsafeLogger = (name: string, options: LoggerOptions = {}): Logger => {
+	const logger = createNewLogger(name, options);
+	return options.fields ? logger.child({ ...options.fields }) : logger;
```

Comment:
Only thing I wonder is if when we search in the Splunk if it will be able to search on the level down.
e.g. if i splunk jiraHost="banana" is it going to search the fields attribute or do I need to explicitly search fields attribute for the jiraHost?🤷‍♂️

Reply:
yeah, sorry, that was old code I didn't removed.  Fixed.